core/shared/src/main/scala/zio/internal/impls/PartitionedLinkedQueue.scala

```diff
@@ -31,7 +31,7 @@ private[zio] final class PartitionedLinkedQueue[A <: AnyRef](
 
   private[this] val mask    = MutableConcurrentQueue.roundToPow2MinusOne(preferredPartitions)
   private[this] val nQueues = mask + 1
-  private[this] val queues  = Array.fill(nQueues)(new LinkedQueue[A](addMetrics = addMetrics))
+  private[this] val queues  = Array.fill(nQueues)(if (addMetrics) new LinkedQueue[A] else new FastLinkedQueue[A])
```

Comment:
Since this is only used in the ZScheduler, in order to avoid the virtual calls on MutableConcurrentQueue, should we just remove the option to add metrics to this queue and use only `FastLinkedQueue` in the partitions?

Reply:
I'm also 2 minds whether `PartitionedLinkedQueue` and `FastLinkedQueue` should extend `MutableConcurrentQueue` at all. Perhaps we don't need to and we can just make the both `private`

Reply:
Yeah I was thinking that too. Will change

Reply:
Done

Reply:
We might actually not even need `FastLinkedQueue`, we can just use `new java.util.concurrent.ConcurrentLinkedQueue[A]()` as the partitions

Reply:
Applied your suggestions